frontend/src/components/Cases/CasesSection.tsx

```diff
@@ -1,20 +1,119 @@
-import { ReportData } from "../../data/parseReportData";
+import { ChangeEvent, useState } from "react";
+import { ReportData, CaseResult } from "../../data/parseReportData";
 import CaseItem from "./CaseItem";
-import { Accordion } from "react-bootstrap";
+import {
+  Accordion,
+  Row,
+  Col,
+  Form,
+  Dropdown,
+  ButtonGroup,
+} from "react-bootstrap";
 
 const CasesSection = ({ reportData }: { reportData: ReportData }) => {
-  const implementations = Array.from(reportData.implementations.values());
+  const [searchText, setSearchText] = useState<string>("");
+  const [filterCriteria, setFilterCriteria] = useState<string[]>([]);
+
+  const handleSearchChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setSearchText(e.target.value.toLowerCase());
+  };
+
+  const handleCheckboxChange = (criteria: string) => {
+    if (filterCriteria.includes(criteria)) {
+      setFilterCriteria(filterCriteria.filter((item) => item !== criteria));
+    } else {
+      setFilterCriteria([...filterCriteria, criteria]);
+    }
+  };
+
+  const filteredCases = Array.from(reportData.cases.entries())
+    .filter(([seq]) => {
+      if (filterCriteria.length === 0) return true;
+      const caseResults: (CaseResult | undefined)[] = Array.from(
+        reportData.implementations.values(),
+      )
+        .map((impl) => impl.cases.get(seq))
+        .flat();
+      return caseResults.some((result) =>
+        filterCriteria.includes(result?.state ?? ""),
+      );
+    })
+    .filter(([, caseData]) => {
+      if (typeof caseData.description !== "string") return false;
+      const regex = new RegExp(searchText, "i");
+      return regex.test(caseData.description);
+    });
+
   return (
-    <Accordion id="cases">
-      {Array.from(reportData.cases.entries()).map(([seq, caseData], index) => (
-        <CaseItem
-          key={index}
-          seq={seq}
-          caseData={caseData}
-          implementations={implementations}
-        />
-      ))}
-    </Accordion>
+    <div>
+      <Row className="mt-3">
+        <Col
+          md={{ span: 6, offset: 6 }}
```

Comment:
input and dropdown share the same space - if the texts on dropdown button changes eveyrhing changes its width slightly. I would put them i separate columns

Reply:
I have tried this, 
![Screenshot (114)](https://github.com/bowtie-json-schema/bowtie/assets/94161758/0986cded-c5af-478b-af6f-3dc635360956)
How is it looking now?

Reply:
For me the layout on last pic looks best!

Reply:
Cool, let's finalize it.